x-pack/solutions/search/packages/kbn-search-queries/src/generate_query.ts

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RetrieverContainer, SearchHighlight } from '@elastic/elasticsearch/lib/api/types';
+
+import {
+  IndexFields,
+  IndicesQuerySourceFields,
+  GenerateQueryOptions,
+  QuerySourceFields,
+} from './types';
+
+const SEMANTIC_FIELD_TYPE = 'semantic';
+
+interface Matches {
+  queryMatches: any[];
+  knnMatches: any[];
+}
+
+export function generateSearchQuery(
```

Comment:
I didn't update `search_playground` to use this code yet, as I didn't want to conflict or break it. But we should plan to migrate to use just this ASAP.